### PR TITLE
Feature/6 fetch most rented books

### PR DESCRIPTION
**Pull request by lukavi44**

### CODE REVIEW

File: library-app/src/components/Sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -36,15 +34,7 @@ const Sidebar = ({ accessToken }: SidebarProps) => {
...
-        {accessToken && (
-          <div className={styles['btn-holder']} onClick={() => setAdminOptions(!adminOptions)}>
-            <button type='button'>
-              <img src={showMore} alt='' />
-            </button>
-          </div>
-        )}
-        {adminOptions && accessToken && (
-          <nav className={styles.sidebar}>
+        { currentUserAdmin(localStorage.getItem('accessToken') || '')  &&
```
Comment (sttevan) [inline library-app/src/components/Sidebar/Sidebar.tsx]: ova logika treba da je u servisu

File: library-app/src/components/UI/BookCarousel.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import styles from './BookCarousel.module.css'
+import { convertDateToString } from '../../helpers/convertDate.helpers';
+import { useNavigate } from 'react-router-dom';
+
+const BookCarousel = () => {
+    const [mostRentedBooks, setMostRentedBooks] = useState<MostRentedBooks[]>([])
+    const navigate = useNavigate()
+    
+    useEffect(() => {
+        getMostRentedBooks(10).then((response) => {
```
Comment (sttevan) [inline library-app/src/components/UI/BookCarousel.tsx]: 10 izvuci u konstantu

File: library-app/src/components/pages/BookDetails/BooksDetails.tsx [hunk 3]
```diff
@@ -110,52 +120,60 @@ const BookDetails = () => {
...
       <div className={styles['container-right']}>
         <img
           src={bookDetails?.Cover ? `data:image/png;base64, ${bookDetails?.Cover}` : placeholder}
           alt='Book Cover'
         />
-        <div className={styles['book-quantity']}>
-          <p>Quantity </p>
-          <p className={styles['book-quantity-number']}>{bookDetails.Quantity}</p>
-        </div>
+        {currentUserAdmin(getAccessToken() || '') &&
```
Comment (sttevan) [inline library-app/src/components/pages/BookDetails/BooksDetails.tsx]: evo ga opet :D

File: library-app/src/components/pages/BookDetails/BooksDetails.tsx [hunk 4]
```diff
@@ -110,52 +120,60 @@ const BookDetails = () => {
...
+            <p className={styles['book-quantity-number']}>{bookDetails.Quantity}</p>
+          </div>}
         <div
           className={
             bookDetails.Available > 0 ? `${styles.available}` : `${styles['available-zero']}`
           }
         >
           <p>Available</p>
-          <p>{bookDetails.Available}</p>
+          {currentUserAdmin(getAccessToken() || '') && <p>{bookDetails.Available}</p>}
```
Comment (sttevan) [inline library-app/src/components/pages/BookDetails/BooksDetails.tsx]: i ovde :D

File: library-app/src/components/pages/BookDetails/BooksDetails.tsx [hunk 5]
```diff
@@ -110,52 +120,60 @@ const BookDetails = () => {
...
-          className={styles['action-btn']}
-          id={styles.edit}
-          onClick={() => {
-            setIsEditModalOpened(true)
-          }}
-        >
-          Edit
-        </button>
+        {
+          currentUserAdmin(getAccessToken() || '') &&
```
Comment (sttevan) [inline library-app/src/components/pages/BookDetails/BooksDetails.tsx]: i ovde

File: library-app/src/components/pages/BookDetails/BooksDetails.tsx [hunk 6]
```diff
@@ -110,52 +120,60 @@ const BookDetails = () => {
...
         )}
-        <button
-          style={{ background: '#eed9c4' }}
-          className={styles['action-btn']}
-          id={styles.delete}
-          onClick={() => setIsDeleteModalOpened(true)}
-        >
-          Delete
-        </button>
+        {currentUserAdmin(getAccessToken() || '') &&
```
Comment (sttevan) [inline library-app/src/components/pages/BookDetails/BooksDetails.tsx]: i ovde

File: library-app/src/components/pages/BookDetails/BooksDetails.tsx [hunk 7]
```diff
@@ -181,13 +200,15 @@ const BookDetails = () => {
             />
           </Modal>
         )}
-        <button onClick={() => setIsRentHistoryModalOpened(true)} className={styles['action-btn']}  style={{ background: '#eed9c4' }} >View History of Rentals</button>
+        {currentUserAdmin(getAccessToken() || '') &&
```
Comment (sttevan) [inline library-app/src/components/pages/BookDetails/BooksDetails.tsx]: i ovde

File: library-app/src/helpers/roles.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import { Jwt, roleKey, UserRole } from '../models/jwt.model';
+// eslint-disable-next-line camelcase
+import jwt_decode from 'jwt-decode';
+
+export const currentUserAdmin = (token: string) => {
```
Comment (sttevan) [inline library-app/src/helpers/roles.tsx]: Necemo token da prosledjujemo ovom, nego neka ovaj servis pozove token servis koji samo procita token is local storage-a

File: library-app/src/helpers/roles.tsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    const decoded: Jwt | '' = token && jwt_decode(token);
+    if (decoded === '') return false
+    return decoded[roleKey] === UserRole.Admin  
+}
+export const currentUserLibrarian = (token: string) => {
```
Comment (sttevan) [inline library-app/src/helpers/roles.tsx]: za sve tri ove metode

**Review by sttevan — CHANGES_REQUESTED** [review]

File: library-app/src/helpers/roles.tsx (hunk 8 above)
Comment (lukavi44) [reply library-app/src/helpers/roles.tsx]: jasno jasno, ima vise smisla :D

File: library-app/src/components/Books/BooksItem/BooksItem.module.css [hunk 10]
```diff
@@ -3,31 +3,52 @@
...
   font-weight: bold;
 }
+
+.book-description {
+  word-break: break-word;
+}
+
+@media screen and (max-width: 426px) {
+  .book-img {
+    /* object-fit: fill; */
```
Comment (SaleBojke) [inline library-app/src/components/Books/BooksItem/BooksItem.module.css]: brisi

File: library-app/src/components/Sidebar/Sidebar.tsx [hunk 11]
```diff
@@ -36,15 +34,7 @@ const Sidebar = ({ accessToken }: SidebarProps) => {
...
-        {accessToken && (
-          <div className={styles['btn-holder']} onClick={() => setAdminOptions(!adminOptions)}>
-            <button type='button'>
-              <img src={showMore} alt='' />
-            </button>
-          </div>
-        )}
-        {adminOptions && accessToken && (
-          <nav className={styles.sidebar}>
+        { currentUserAdmin()  &&
```
Comment (SaleBojke) [inline library-app/src/components/Sidebar/Sidebar.tsx]: what is with role librarian?

File: library-app/src/App.module.css
Comment (vladimir-agbaba-valcon) [inline library-app/src/App.module.css:1]: If this file is empty now, you can delete it

File: library-app/src/components/Sidebar/Sidebar.tsx [hunk 12]
```diff
@@ -2,18 +2,16 @@ import React, { useState } from 'react'
 import styles from './Sidebar.module.css'
 import home from '../../assets/icons/home.png'
 import account from '../../assets/icons/account.png'
-import showMore from '../../assets/icons/showMore.png'
 import { useNavigate } from 'react-router-dom'
 import Modal from '../Layout/Modal'
 import ManageBookForm from '../Books/BooksList/ManageBookForm'
-
+import { currentUserAdmin} from '../../helpers/roles'
```
Comment (vladimir-agbaba-valcon) [inline library-app/src/components/Sidebar/Sidebar.tsx]: Add empty line before interface

File: library-app/src/components/pages/Login/Login.module.css [hunk 13]
```diff
@@ -1,6 +1,6 @@
 .container {
   background-color: #98a0a6;
-  height: 32rem;
+  /* height: 32rem; */
```
Comment (vladimir-agbaba-valcon) [inline library-app/src/components/pages/Login/Login.module.css]: Commented code

File: library-app/src/components/UI/BookCarousel.tsx [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+import { convertDateToString } from '../../helpers/convertDate.helpers';
+import { useNavigate } from 'react-router-dom';
+
+const BookCarousel = () => {
+    const [mostRentedBooks, setMostRentedBooks] = useState<MostRentedBooks[]>([])
+    const rentCount = 10;
+    const navigate = useNavigate()
+    
+    useEffect(() => {
+        getMostRentedBooks(rentCount).then((response) => {
```
Comment (SaleBojke) [inline library-app/src/components/UI/BookCarousel.tsx:16]: ajde da ne udaramo ovaj request ako nije ulogovan admin, jer ce nam podaci biti prisutni u state-u bespotrebno.

File: library-app/src/helpers/roles.tsx [hunk 15]
```diff
@@ -0,0 +1,23 @@
+import { Jwt, roleKey, UserRole } from '../models/jwt.model';
+// eslint-disable-next-line camelcase
```
Comment (vladimir-agbaba-valcon) [inline library-app/src/helpers/roles.tsx:2]: Maybe you can solve this by typing import jwt_decode as JwtDecode from 'jwt-decode'; but leave it like this for now

**Review by vladimir-agbaba-valcon — APPROVED** [review]

File: library-app/src/components/pages/BookDetails/BooksDetails.tsx [hunk 16]
```diff
@@ -110,52 +118,60 @@ const BookDetails = () => {
...
-        <img
+        <img className={styles['book-img']}
           src={bookDetails?.Cover ? `data:image/png;base64, ${bookDetails?.Cover}` : placeholder}
           alt='Book Cover'
         />
-        <div className={styles['book-quantity']}>
-          <p>Quantity </p>
-          <p className={styles['book-quantity-number']}>{bookDetails.Quantity}</p>
-        </div>
+        {currentUserAdmin() &&
```
Comment (SaleBojke) [inline library-app/src/components/pages/BookDetails/BooksDetails.tsx]: librarian?

File: library-app/src/components/pages/Login/Login.module.css (hunk 13 above)
Comment (SaleBojke) [inline library-app/src/components/pages/Login/Login.module.css]: brisi

File: library-app/src/components/pages/BookDetails/BooksDetails.tsx [hunk 17]
```diff
@@ -110,52 +118,60 @@ const BookDetails = () => {
...
         )}
-        <button
-          style={{ background: '#eed9c4' }}
-          className={styles['action-btn']}
-          id={styles.delete}
-          onClick={() => setIsDeleteModalOpened(true)}
-        >
-          Delete
-        </button>
+        {currentUserAdmin() &&
```
Comment (SaleBojke) [inline library-app/src/components/pages/BookDetails/BooksDetails.tsx]: isto

File: library-app/src/components/pages/BookDetails/BooksDetails.tsx [hunk 18]
```diff
@@ -181,13 +198,15 @@ const BookDetails = () => {
             />
           </Modal>
         )}
-        <button onClick={() => setIsRentHistoryModalOpened(true)} className={styles['action-btn']}  style={{ background: '#eed9c4' }} >View History of Rentals</button>
+        {currentUserAdmin() &&
```
Comment (SaleBojke) [inline library-app/src/components/pages/BookDetails/BooksDetails.tsx]: isto

**Pull request merged**